doc/sphinx/grammar.rst

```diff
@@ -0,0 +1,2247 @@
+                 : inversion_clear `quantified_hypothesis` `as_or_and_ipat` `in_hyp_list`
+                 : inversion `quantified_hypothesis` using `constr` `in_hyp_list`
+                 : red `clause_dft_concl`
+                 : hnf `clause_dft_concl`
+                 : simpl `delta_flag` `ref_or_pattern_occ_opt` `clause_dft_concl`
+                 : cbv `strategy_flag` `clause_dft_concl`
+                 : cbn `strategy_flag` `clause_dft_concl`
+                 : lazy `strategy_flag` `clause_dft_concl`
+                 : compute `delta_flag` `clause_dft_concl`
+                 : vm_compute `ref_or_pattern_occ_opt` `clause_dft_concl`
+                 : native_compute `ref_or_pattern_occ_opt` `clause_dft_concl`
+                 : unfold `unfold_occ_list` `clause_dft_concl`
+                 : fold `constr_list` `clause_dft_concl`
+                 : pattern `pattern_occ_list` `clause_dft_concl`
+                 : change `conversion` `clause_dft_concl`
+                 : btauto
+                 : rtauto
+                 : congruence
+                 : congruence `integer`
+                 : congruence with `constr_list`
+                 : congruence `integer` with `constr_list`
+                 : f_equal
+                 : firstorder `tactic_opt` `firstorder_using`
+                 : firstorder `tactic_opt` with `preident_list`
+                 : firstorder `tactic_opt` `firstorder_using` with `preident_list`
+                 : gintuition `tactic_opt`
+                 : functional inversion `quantified_hypothesis` `reference_opt`
+                 : functional induction `constr_list` `fun_ind_using` `with_names`
+                 : soft functional induction `constr_list` `fun_ind_using` `with_names`
+                 : myred
+                 : psatz_Z `int_or_var` `tactic`
+                 : psatz_Z `tactic`
+                 : xlia `tactic`
+                 : xnlia `tactic`
+                 : xnra `tactic`
+                 : xnqa `tactic`
+                 : sos_Z `tactic`
+                 : sos_Q `tactic`
+                 : sos_R `tactic`
+                 : lra_Q `tactic`
+                 : lra_R `tactic`
+                 : psatz_R `int_or_var` `tactic`
+                 : psatz_R `tactic`
+                 : psatz_Q `int_or_var` `tactic`
+                 : psatz_Q `tactic`
+                 : nsatz_compute `constr`
+                 : omega
+                 : omega with `ident_list`
+                 : omega with *
+                 : protect_fv `string` in `ident`
+                 : protect_fv `string`
+                 : ring_lookup `tactic_expr0` [ `constr_list_opt` ] `constr_list`
+                 : field_lookup `tactic` [ `constr_list_opt` ] `constr_list`
+                 : YouShouldNotTypeThis `ssrintrosarg`
+                 : by `ssrhintarg`
+                 : YouShouldNotTypeThis do `ssrdoarg`
+                 : YouShouldNotTypeThis `ssrtclarg` `ssrseqdir` `ssrseqarg`
```

Comment:
:laughing: :rofl:

Reply:
The worst part of this is that the fix for this hack is pretty trivial now that we have a proper API to define generic arguments.

Reply:
I think @maximedenes had it is his drawer at some point. The problem was that ARGUMENT EXTEND would not accept 0 productions, so I had to write a dummy one.